### DISCUSSION TITLE
The process has been signaled with signal "4"

### DISCUSSION THREAD

**Discussion by eng-faris** [spatie/browsershot#620] [Q&A]
I use cents 8 and node v10.24.0 I fallow https://frugalisminds.com/how-to-setup-puppeteer-in-centos-7/
but still problem.

**Comment by freekmurze** [comment]
We don't have access to such an environment. Feel free to submit a PR that fixes this for you.

**Comment by eng-faris** [comment]
How?

**Comment by tempKaew** [comment]
# Browsershot Signal 4 Error due to SELinux

### **Problem**

When using `laravel browsershot`, you encounter the error: **`The process has been signaled with signal "4"`**. This issue is typically caused by **SELinux** (Security-Enhanced Linux) on the server blocking the **Node.js** process from using executable memory (`execmem`).

This often occurs when Node.js is invoked by a web server like Apache or Nginx, which runs under the strict `httpd_t` SELinux context.

### **Cause**

By default, SELinux policies restrict processes running under the `httpd_t` context from using `execmem` for security reasons. However, certain Node.js operations, such as JIT compilation, require this permission. SELinux identifies this as a potential security violation and blocks the action, causing the Node.js process to terminate with signal 4.

### **Solution**

1.  **Check SELinux Logs**
    To confirm that SELinux is the root cause, check the audit logs with the following command:

    ```bash
    sudo ausearch -m avc -ts recent
    ```

    Alternatively, you can monitor the log file directly:

    ```bash
    sudo tail -f /var/log/audit/audit.log
    ```

    If you see a message similar to `avc: denied { execmem } for ... comm="node"`, it confirms the issue is with SELinux.

2.  **Allow `execmem` for `httpd_t` (Recommended Solution)**
    Permanently enable the `httpd_execmem` boolean to allow this behavior. The `-P` flag makes the change persistent across reboots.

    ```bash
    sudo setsebool -P httpd_execmem 1
    ```

    This command allows the web server to properly execute Node.js processes that require `execmem`.
